### PR TITLE
Deduplicate some block-to-JSON code

### PR DESCRIPTION
**Pull request by domob1812**
Some of the logic converting blocks and block headers to JSON for the blockchain RPC methods (`getblock`, `getlockheader`) was duplicated. Instead of that, the `blockToJSON` RPC method now calls `blockheaderToJSON` first, and then fills in the missing, block-specific bits explicitly.

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -156,21 +156,14 @@ UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex
 
 UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIndex* blockindex, bool txDetails)
 {
-    // Serialize passed information without accessing chain state of the active chain!
-    AssertLockNotHeld(cs_main); // For performance reasons
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:160]: why is this removed?

Comment (domob1812) [reply src/rpc/blockchain.cpp:160]: Because it is also part of `blockheaderToJSON`, and that is also clear immediately to readers of the code in this case.

But I was thinking about whether or not to remove it myself and was not sure.  So if you think it should be there (even though it will be asserted again in `blockheaderToJSON`), I'm happy to put it back.

**Comment by 0xB10C** [comment]
ACK 1c5474883836bcadbb7f649ef9bf95137d991e49. Tested that the contents of the `getblock` JSON response of master and this PR are equal (for two testnet blocks; one older and a recent block). The ordering of the JSON fields does change, but I don't think that's a problem. 

Reviewers might want to use `jq -S` to sort the JSON fields alphabetically and then compare the RPC responses. Additionally, mainnet and a broader range of blocks could be tested.

**Comment by 0xB10C** [comment]
Checked the `rest/block/<hash>.json` interface too as it calls `blockToJSON()` as well.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -156,21 +156,12 @@ UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex
...
 
-    UniValue result(UniValue::VOBJ);
-    result.pushKV("hash", blockindex->GetBlockHash().GetHex());
-    const CBlockIndex* pnext;
-    int confirmations = ComputeNextBlockAndDepth(tip, blockindex, pnext);
-    result.pushKV("confirmations", confirmations);
     result.pushKV("strippedsize", (int)::GetSerializeSize(block, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS));
     result.pushKV("size", (int)::GetSerializeSize(block, PROTOCOL_VERSION));
     result.pushKV("weight", (int)::GetBlockWeight(block));
     result.pushKV("height", blockindex->nHeight);
```
Comment (laanwj) [inline src/rpc/blockchain.cpp:164]: It looks like `height` is also set upstream?

Comment (domob1812) [reply src/rpc/blockchain.cpp]: Yes indeed, good catch!  I've removed that line as well now.

**Comment by laanwj** [comment]
Code review ACK fa2c52111544b0de93ac1002f5395bceeb8fea0e
